src/converse-service-administration.js

```diff
@@ -0,0 +1,412 @@
+                    node: this.model.attributes.command_node
+                }).up();
+
+                _converse.connection.sendIQ(stanza, result => {
+                    const session_id = sizzle('command', result).pop().getAttribute('sessionid');
+                    const fields = sizzle('field', result);
+                    const form_type_value = sizzle('field[var="FORM_TYPE"] > value', result).pop().innerHTML;
+
+                    const attributes = {
+                        'id': _converse.SERVICE_ADMIN_TYPE,
+                        'jid': _converse.domain,
+                        'type': _converse.SERVICE_ADMIN_TYPE,
+                        'from': _converse.domain,
+                        'fields': Array.from(fields,
+                                             field => this.getAttributesFromField(field)),
+                        'form_type_value': form_type_value,
+                        'name': this.model.get('name'),
+                        'node': this.model.get('command_node'),
+                        'header_text': this.model.attributes.description,
+                        'send_button_text': __('send'),
+                        'placeholder_text': this.model.attributes.description,
+                        'session_id': session_id
+                    };
+
+                    const existing_chatbox = _converse.chatboxes.get(_converse.SERVICE_ADMIN_TYPE);
+                    if (existing_chatbox) {
+                        existing_chatbox.set(attributes);
+                        existing_chatbox.maybeShow(true);
+                    } else {
+                        const chatbox = new _converse.ServiceAdminBox(attributes);
+
+                        _converse.chatboxes.add(chatbox);
+                        chatbox.maybeShow(true);
+                    }
+                }, error => {
+                    _converse.log(error, Strophe.LogLevel.ERROR);
+                });
+            },
+
+            getAttributesFromField (field) {
+                const attribute_names = field.getAttributeNames();
+                const obj = {};
+                attribute_names.forEach(name => {
+                    obj[name] = field.getAttribute(name);
+                });
+                obj["value"] = this.getValueOfFieldElement(field);
+                obj["required"] = field.getElementsByTagName('required').length !== 0;
+                if (obj.var === "body" && obj.value !== null && obj.value.length === 0) {
+                    obj.value = __("-- sent from %1$s",
+                            Strophe.getBareJidFromJid(_converse.jid));
+                }
+                return obj;
+            },
+
+            getValueOfFieldElement (field) {
+                const values = field.getElementsByTagName('value');
+                if (values.length > 0) {
+                    return values[0].innerHTML;
+                } else {
+                    return '';
+                }
+            },
+        });
```

Comment:
This looks like a utility method which you can put in src/headless/utils/form.js and then call in this plugin via `u.getValueOfFieldElement`.

Make sure to import form.js in this plugin if you don't already.